chart/compass/charts/instance-creator/templates/deployment.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Chart.Name }}
+    release: {{ .Release.Name }}
+spec:
+  replicas: {{ .Values.deployment.minReplicas }}
+  selector:
+    matchLabels:
+      app: {{ .Chart.Name }}
+      release: {{ .Release.Name }}
+  strategy:
+    {{- toYaml .Values.deployment.strategy | nindent 4 }}
+  template:
+    metadata:
+      annotations:
+        restartOnNewInstallationMarker: {{ randAlphaNum 5 | quote }} # Restarts the deployment on a new Helm installation. (https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments)
```

Comment:
I saw that it is used in other deployments and applied it here but haven't dug into it.  By memory, it was related to able to restart components when new installation is triggered so that the secrets can be refreshed. I don't remember the semantics of the annotation, though.

However, I also can't find info about it.

We can investigate it further in the team to double check if this functionality still works and if needed search/validate a new approach and apply it across all the resources in the project. 

What do you think?